### ISSUE TITLE
[newFeature] setFavoritePokemon

### ISSUE THREAD

**Issue by SonnyX** [Grover-c13/PokeGOAPI-Java#219]
**Description:**
Add a method to set the Pokemon as a favorite.

**Expected behavior:**
When true it becomes favorite, when false it clears it.

Should be placed at Pokemon.java, please add it without breaking checkStyleMain :)

```
    /**
    * Function to mark the pokemon as favorite or not.
    *
    * @param markFavorite Mark Pokemon as Favorite?
    * @return the SetFavoritePokemonResponse.Result
    * @throws LoginFailedException  the login failed exception
    * @throws RemoteServerException the remote server exception
    */
    public SetFavoritePokemonResponse.Result setFavoritePokemon(boolean markFavorite) 
            throws LoginFailedException, RemoteServerException {
        SetFavoritePokemonMessage reqMsg = SetFavoritePokemonMessage.newBuilder()
                .setPokemonId(getId())
                .setIsFavorite(markFavorite)
                .build();

        ServerRequest serverRequest = new ServerRequest(RequestType.SET_FAVORITE_POKEMON, reqMsg);
        pgo.getRequestHandler().sendServerRequests(serverRequest);

        SetFavoritePokemonResponse response;
        try {
            response = SetFavoritePokemonResponse.parseFrom(serverRequest.getData());
        } catch (InvalidProtocolBufferException e) {
            throw new RemoteServerException(e);
        }

        pgo.getInventories().getPokebank().removePokemon(this);
        pgo.getInventories().updateInventories();

        return response.getResult();
    }
```

**Comment by Jari27** [comment]
If you tested it and it works, why not submit a pull request?

**Comment by SonnyX** [comment]
Couldn't get it to pass checkStyleMain at first, now locally it succeeds. #222

**Comment by Jari27** [comment]
Why do you need to call `pgo.getInventories().getPokebank().removePokemon(this);`? 

Also, maybe add a function isFavorite()?

**Comment by SonnyX** [comment]
@Sadye I already saw a function named getFavorite() with most likely the same bahavior.
Also I called the `removePokemon(this);` you named above because the similar function `renamePokemon(NickName)` uses it as well, didn't put too much thought into it.

**Comment by vmarchaud** [comment]
`getFavorite()` is just a `isFavorite()`, dont hesite to make a PR to deprecate `getFavorite()` and create a `isFavorite()`.
since the method is merged in #222 i close the issue.
